### PR TITLE
ENH: stats: Add implementation of _entropy for the t distr.

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
Inspired by gh-7542 and gh-13233.

### CODE REVIEW

File: scipy/stats/tests/test_distributions.py [hunk 1]
```diff
@@ -1966,6 +1966,14 @@ def test_moments_t():
     assert_equal(stats.t.stats(df=4.01, moments='sk'), (0.0, 6.0/(4.01 - 4.0)))
 
 
+def test_t_entropy():
+    df = [1, 2, 25, 100]
+    # Expected values were computed with mpmath.
+    expected = [2.5310242469692907, 1.9602792291600821,
+                1.459327578078393, 1.4289633653182439]
+    assert_allclose(stats.t.entropy(df), expected, rtol=1e-14)
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:1974]: Checked:
```
from mpmath import mp, mpf, digamma, beta, log, sqrt
mp.dps = 20
nu = 100
print((nu+1)/2 * (digamma((1+nu)/2) - digamma(nu/2)) + log(sqrt(nu) * beta(0.5, nu/2)))
```

File: scipy/stats/_continuous_distns.py [hunk 2]
```diff
@@ -6015,6 +6015,12 @@ def _stats(self, df):
         g2 = np.where(df <= 2, np.nan, g2)
         return mu, mu2, g1, g2
 
+    def _entropy(self, df):
+        half = df/2
+        half1 = (df + 1)/2
+        return (half1*(sc.digamma(half1) - sc.digamma(half)) +
+                np.log(np.sqrt(df)*sc.beta(half, 0.5)))
```
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:6022]: Looks good.

**Review by mdhaber — APPROVED** [review]
LGTM

File: scipy/stats/_continuous_distns.py (hunk 2 above)
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:6022]:
```suggestion
        return (half1*(sc.digamma(half1) - sc.digamma(half))
                + np.log(np.sqrt(df)*sc.beta(half, 0.5)))
```

Comment (WarrenWeckesser) [reply scipy/stats/_continuous_distns.py]: Committed.  I've always had a slight preference for the old style, but Knuth-style is the current recommendation, so +1.

**Comment by WarrenWeckesser** [comment]
Thanks @mdhaber.

The one test on Travis CI that failed is not related to this PR.  It is the old sporadic failure of the ARPACK test `test_symmetric_nodes`.

**Pull request merged**